libs/core/kiln_ai/datamodel/external_tool_server.py

```diff
@@ -84,9 +84,6 @@ def validate_required_fields(self) -> "ExternalToolServer":
                     raise ValueError(
                         "env_vars must be a dictionary for external tools of type 'local_mcp'"
                     )
-                # Set the default value if not provided
-                if "env_vars" not in self.properties:
-                    self.properties["env_vars"] = {}
```

Comment:
Don't set values in validators. Treat them as strictly validators (They should be read only). Just require it and callers will figure it out.